client/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/crypto/LocalClientCryptoServiceImpl.java

```diff
@@ -1,33 +1,415 @@
+
+            cryptoResponseDto.setValue(base64encoder.encodeToString(encrypted_key_iv_data));
+            return cryptoResponseDto;
+        } catch(Exception e) {
+            Log.e(TAG, "encrypt: Encryption failed ", e);
+        }
+
         return null;
     }
 
     @Override
     public CryptoResponseDto decrypt(CryptoRequestDto cryptoRequestDto) {
+        CryptoResponseDto cryptoResponseDto = new CryptoResponseDto();
+        byte[] dataToDecrypt =  base64decoder.decode(cryptoRequestDto.getValue());
+        byte[] public_key =  base64decoder.decode(cryptoRequestDto.getPublicKey());
+
+        byte[] encryptedSecretKey = Arrays.copyOfRange(dataToDecrypt, 0, KEYGEN_SYMMETRIC_KEY_LENGTH);
+        byte[] iv = Arrays.copyOfRange(dataToDecrypt, KEYGEN_SYMMETRIC_KEY_LENGTH, KEYGEN_SYMMETRIC_KEY_LENGTH+CRYPTO_SYMMETRIC_IV_LENGTH);
+        byte[] aad = Arrays.copyOfRange(dataToDecrypt,KEYGEN_SYMMETRIC_KEY_LENGTH+CRYPTO_SYMMETRIC_IV_LENGTH, KEYGEN_SYMMETRIC_KEY_LENGTH+CRYPTO_SYMMETRIC_IV_LENGTH+CRYPTO_SYMMETRIC_AAD_LENGTH);
+        byte[] encrypted_data = Arrays.copyOfRange(dataToDecrypt, KEYGEN_SYMMETRIC_KEY_LENGTH+CRYPTO_SYMMETRIC_IV_LENGTH+CRYPTO_SYMMETRIC_AAD_LENGTH, dataToDecrypt.length);
+
+        try {
+            PrivateKey privateKey = getEnDecPrivateKey();
+
+            // asymmetric decryption of secret key----------------------------------------------------
+            final Cipher cipher_asymmetric = Cipher.getInstance(CRYPTO_ASYMMETRIC_ALGORITHM);
+            cipher_asymmetric.init(Cipher.DECRYPT_MODE, privateKey);
+            byte[] secretKeyBytes = cipher_asymmetric.doFinal(encryptedSecretKey);
+
+            SecretKey secretKey = new SecretKeySpec(secretKeyBytes, KEYGEN_SYMMETRIC_ALGORITHM);
+
+            // symmetric decryption of data-----------------------------------------------------
+            final Cipher cipher_symmetric = Cipher.getInstance(CRYPTO_SYMMETRIC_ALGORITHM);
+            final GCMParameterSpec gcmParameterSpec = new GCMParameterSpec(CRYPTO_GCM_TAG_LENGTH, iv);
+            cipher_symmetric.init(Cipher.DECRYPT_MODE, secretKey, gcmParameterSpec);
+            cipher_symmetric.updateAAD(aad);
+            byte[] decodedBytes = cipher_symmetric.doFinal(encrypted_data);
+            String decodedString = new String(decodedBytes);
+            cryptoResponseDto.setValue(decodedString);
+            return cryptoResponseDto;
+        } catch(Exception e) {
+            Log.e(TAG, "decrypt: Decryption failed", e);
+        }
         return null;
     }
 
     @Override
     public PublicKeyResponseDto getPublicKey(PublicKeyRequestDto publicKeyRequestDto) {
+        PublicKeyResponseDto publicKeyResponseDto = new PublicKeyResponseDto();
+
+        try {
+            String keyRequest = publicKeyRequestDto.getServerProfile();
+            Log.i(TAG, "getPublicKey: Public key accessed for " + keyRequest);
+            if(keyRequest == "sign") {
```

Comment:
Create a separate class to create these constants. and use static way of accessing this constants.
check this example - https://github.com/mosip/keymanager/blob/develop/kernel/kernel-keymanager-service/src/main/java/io/mosip/kernel/cryptomanager/constant/CryptomanagerConstant.java

and also error codes example from here - https://github.com/mosip/keymanager/blob/develop/kernel/kernel-keymanager-service/src/main/java/io/mosip/kernel/cryptomanager/constant/CryptomanagerErrorCode.java